client/azure_client_set.go

```diff
@@ -152,6 +147,7 @@ func newDeploymentsClient(config *clientConfig) (*resources.DeploymentsClient, e
 	c := resources.NewDeploymentsClientWithBaseURI(config.resourceManagerEndpoint, config.subscriptionID)
 	c.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
 	c.RetryAttempts = 1
+	senddecorator.ConfigureClient(&gatekeeper.Gatekeeper{}, &c)
```

Comment:
Just want to make sure: the Azure rate limit is per client? If that's the case, it makes sense to create a gatekeeper instance for every client, otherwise sharing a single instance would make more sense.

Reply:
No, rate limit is per subscription and ideally the `gatekeeper` should be shared. That's why I raised it to `ConfigureClient()` level. I still want to figure though how to best wire it from the highest level.

The problem here is that different services have different rate-limit quotas but there's only single `client.NewAzureClientset()` call in `resource_set.go`.